http-client-shared-data/src/main/java/org/triplea/http/client/moderator/toolbox/ModeratorToolboxFeignClient.java

```diff
@@ -0,0 +1,69 @@
+package org.triplea.http.client.moderator.toolbox;
+
+import java.util.List;
+import java.util.Map;
+
+import org.triplea.http.client.HttpConstants;
+
+import feign.HeaderMap;
+import feign.Headers;
+import feign.Param;
+import feign.RequestLine;
+
+/**
+ * Http client for moderator 'toolbox' actions. The toolbox is essentially a set of windows
+ * that show lobby table data and provide CRUD operations to moderators. For example
+ * with the toolbox moderators should be able to update the list of restricted usernames,
+ * view players that have joined the lobby, add and remove player bans, etc.
+ * <p>
+ * This set of http methods should all require an API key passed via headers to authenticate
+ * the moderator issuing the request.
+ * </p>
+ * <p>
+ * Rate-limiting: of note, the backend implementation should be careful to apply rate limiting
+ * to any/all endpoints that take an API key so as to avoid brute-force attacks to try and crack
+ * an API key value.
+ * </p>
+ */
+interface ModeratorToolboxFeignClient {
+  @RequestLine("POST " + ModeratorToolboxClient.VALIDATE_API_KEY_PATH)
+  @Headers({HttpConstants.CONTENT_TYPE_JSON, HttpConstants.ACCEPT_JSON})
+  String validateApiKey(@HeaderMap Map<String, Object> headerMap);
+
+  // TODO: test-me
+  @RequestLine("POST " + ModeratorToolboxClient.REGISTER_API_KEY_PATH)
+  @Headers({HttpConstants.CONTENT_TYPE_JSON, HttpConstants.ACCEPT_JSON})
+  RegisterApiKeyResult registerKey(RegisterApiKeyParam registerApiKeyParam);
+  // @Param(ModeratorToolboxClient.SINGLE_USE_KEY_PARAM) String singleUseKey,
+  // @Param(ModeratorToolboxClient.NEW_API_KEY_PASSWORD_PARAM) String newPassword);
```

Comment:
Code comment

Reply:
Not terribly surprised, only the code outside of toolbox or in the DAO layer is really ready for this detailed of code improvement/cleanup. I'd urge you back up on the review magnifying class, try to focus on if we have all the right data, if these are the right updates and not that the code is perfect. With that said, I do appreciate the head start on the review.